src/tests/multihost/alltests/test_default_debug_level.py

```diff
@@ -13,38 +13,66 @@ class TestDefaultDebugLevel(object):
     @pytest.mark.tier1
     def test_0001_check_default_debug_level(self, multihost, backupsssdconf):
         """
-        @Title: IDM-SSSD-TC: Check default debug level by authenticating the
-        user
+        @Title: IDM-SSSD-TC: Check default debug level when sssd start
+        successfully
         """
         section = "domain/%s" % ds_instance_name
         domain_params = {'debug_level': ''}
         tools = sssdTools(multihost.client[0])
         tools.sssd_conf(section, domain_params, action='delete')
+        # stop sssd, delete sssd logs and cache, start sssd
+        tools.clear_sssd_cache()
+        log_list = ['sssd', 'sssd_example1', 'sssd_nss',
+                    'sssd_pam', 'sssd_implicit_files']
+        for log_filename in log_list:
+            log = '/var/log/sssd/%s.log' % log_filename
+            log_str = multihost.client[0].get_file_contents(log).decode(
+                'utf-8')
+            debug_level = ['0x0080', '0x0100', '0x0200', '0x0400', '0x1000',
+                           '0x2000', '0x4000', '0x10000']
+            for level in debug_level:
+                regex = r'%s' % level
+                if re.compile(regex).search(log_str):
```

Comment:
This version is much better, thank you.

But I think there is still a room for an improvement.
This is very inefficient to run a regex for every log level (and btw error prone: 0x8000 level is missing in the list)
I would propose to lookup for "] (%x): " pattern and make sure "x > 0x40"

Reply:
I don't see 0x8000 level in sssd.conf or sssd man page.
To add this condition, 'pattern and make sure "x > 0x40', first we need to find out all debug_level in all logs, then after storing it in a variable, or in the list we can compare that values. To compare hexadecimal, first each value to convert it into integer and then we can compare it. And anyways we will not get any other values in logs other than define values.